harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -143,18 +143,51 @@ def _set_data_loaders(self) -> None:
     def run(self) -> None:
         for w, args, response_func in self.workloads:
             if w.kind == workload.Workload.Kind.RUN_STEP:
-                response_func(
-                    util.wrap_metrics(
-                        self._train_for_step(w.step_id, w.num_batches, w.total_batches_processed),
-                        self.context.get_stop_requested(),
+                try:
+                    response_func(
+                        util.wrap_metrics(
+                            self._train_for_step(
+                                w.step_id, w.num_batches, w.total_batches_processed
+                            ),
+                            self.context.get_stop_requested(),
+                            self.context.get_invalid_hp(),
+                        )
                     )
-                )
+                except det.InvalidHP as e:
+                    logging.info(
+                        "Invalid hyperparameter exception in trial train step: {}".format(e)
+                    )
+                    response_func(
+                        util.wrap_metrics(
+                            self._train_for_step(
+                                w.step_id, w.num_batches, w.total_batches_processed
+                            ),
+                            self.context.get_stop_requested(),
+                            self.context.set_invalid_hp(True),
+                        )
+                    )
+                    raise
             elif w.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-                response_func(
-                    util.wrap_metrics(
-                        self._compute_validation_metrics(), self.context.get_stop_requested()
+                try:
+                    response_func(
+                        util.wrap_metrics(
+                            self._compute_validation_metrics(),
+                            self.context.get_stop_requested(),
+                            self.context.get_invalid_hp(),
+                        )
                     )
-                )
+                except det.InvalidHP as e:
+                    logging.info(
+                        "Invalid hyperparameter exception in trial validation step: {}".format(e)
+                    )
+                    response_func(
+                        util.wrap_metrics(
+                            self._compute_validation_metrics(),
+                            self.context.get_stop_requested(),
+                            self.context.set_invalid_hp(True),
```

Comment:
I think you actually intended to just put a `True` here rather than `self.context.set_invalid_hp(True),`